aptos-move/aptos-sdk-builder/src/golang.rs

```diff
@@ -713,22 +715,21 @@ func decode_{0}_argument(arg aptostypes.TransactionArgument) (value {1}, err err
 
     fn quote_type(type_tag: &TypeTag) -> String {
         use TypeTag::*;
+        let str_tag: StructTag = StructTag::from_str("0x1::string::String").unwrap();
         match type_tag {
             Bool => "bool".into(),
             U8 => "uint8".into(),
             U64 => "uint64".into(),
             U128 => "serde.Uint128".into(),
             Address => "aptostypes.AccountAddress".into(),
-            Vector(type_tag) => match type_tag.as_ref() {
-                Bool => "[]bool".into(),
-                U8 => "[]byte".into(),
-                U64 => "[]uint64".into(),
-                U128 => "[]uint128".into(),
-                Address => "[]aptostypes.AccountAddress".into(),
-                Vector(type_tag) if type_tag.as_ref() == &U8 => "[][]byte".into(),
+            Vector(type_tag) => {
+                format!("[]{}", Self::quote_type(type_tag))
+            }
+            Struct(struct_tag) => match struct_tag.clone() {
+                tag if tag == str_tag => "[]uint8".into(),
```

Comment:
nit: []byte maybe more ergonomic in golang

Reply:
updated